### DISCUSSION TITLE
Quick single server PoC try -- failed on UE attach

### DISCUSSION THREAD

**Discussion by jni2000** [herlesupreeth/docker_open5gs#138] [Q&A]
@herlesupreeth Thanks for your great effort in containerizing this great implementation. I am following instructions in https://github.com/herlesupreeth/docker_open5gs and successfully build and run the whole set of core and IMS contianers. I also started the zmq eNB and zmq LTE UE. Added subscriber via Open5gs web UI (http://localhost:3000/) with all necessary parameters of UE1 as specified in .env. I got attach rejection with cause code 03. Logs in mme showed IMSI not found. What could I do wrong ?

My use cases are as below:
Run core, eNB and UE containers on the same server

~/docker_open5gs$ docker-compose up
~/docker_open5gs$ docker-compose -f srsenb_zmq.yaml up -d && docker attach srsenb_zmq
~/docker-compose -f srsue_zmq.yaml up -d && docker attach srsue_zmq

By the way, I did run "set -a" and "source .env" in all three shells issuing the above docker-compose commands.

**Comment by herlesupreeth** [comment]
Hi @jni2000 since i added support for sending SMS over SGs interface towards MSC (osmomsc) you need to add UE IMSI and MSISDN in osmohlr. please find the instructions to add them in below link

https://github.com/s5uishida/docker_open5gs_volte_sms_config#register-the-imsi-and-msisdn-with-osmohlr

**Reply by jni2000** [reply]
Thanks, @herlesupreeth, for your reply. I followed your instructions as below, still got the same attach failure.

1. added IMSI and MSISDN in osmohlr as shown below:

```
OsmoHLR# show subscribers all
ID     MSISDN        IMSI              IMEI              NAM
-----  ------------  ----------------  ----------------  -----
1      33611123456   208930100001111   356113022094149   CSPS  
2      1001          001010000001001    -------------    CSPS  
```
2. My Open5GS subscriber info are as below (I tried to match what are mentioned in your instructions, but the webui seems default to what I have. Do not know how to fix this)
![Screenshot from 2022-12-28 11-21-25](https://user-images.githubusercontent.com/11164076/209841995-c4374e65-0069-42d2-b502-3efe13e5a1ba.png)

3. And my FHoSS are as below

![Screenshot from 2022-12-28 11-23-54](https://user-images.githubusercontent.com/11164076/209842344-c9544e18-e177-4908-bd78-32f50995ecdc.png)
![Screenshot from 2022-12-28 11-24-20](https://user-images.githubusercontent.com/11164076/209842345-3a172794-09d0-49a3-a4bb-993601cf3445.png)
![Screenshot from 2022-12-28 11-24-41](https://user-images.githubusercontent.com/11164076/209842346-59defb01-086c-4172-a11a-9d717b3d154f.png)

**Comment by herlesupreeth** [comment]
I see two plmn in your osmohlr and also please re-read the instructions for how to provision user in open5gs and fhoss as they are incorrect.

once you have correctly provisioned please take a trace (.pcap) file of the issue you are facing

**Reply by jni2000** [reply]
thanks @herlesupreeth. are you interested in providing consulting service to us? We are a US company looking to evaluate LET/5G core solutions. Thanks.

**Comment by herlesupreeth** [comment]
@jni2000 Thank you for the offer. Unfortunately, I don't have the bandwidth to offer consulting services.

**Reply by jni2000** [reply]
Thanks, @herlesupreeth. No problem. In our case, we only need to run the 5gs cores and we do not need the IMS and MSC etc. elements, what is the quickest way to customize your docker compose and configuration here ?

**Comment by s5uishida** [comment]
Hi @jni2000 

First of all, how about simulating gNodeB and UE using UERANSIM and confirming the operation of UE-RAN-5GC? (most easist)
After that, I think it would be good to confirm the operation with real gNodeB and UE.

**1) Edit the `.env` file.**
Set the following parameter according to your environment. Other defaults are fine.
```
DOCKER_HOST_IP=192.168.1.223
```
**2) Build Docker images.**
```
cd docker_open5gs/base
docker build --no-cache --force-rm -t docker_open5gs .
...
cd ../ueransim
docker build --no-cache --force-rm -t docker_ueransim .
```
**3) Start Open5GS 5G SA.**
```
cd docker_open5gs
set -a
source .env
docker-compose -f sa-deploy.yaml up
```
**4) Register sample subscriber information.**
```
http://[DOCKER_HOST_IP]:3000/
username: admin
password: 1423
```
```
UE1_IMSI=001011234567895
UE1_KI=8baf473f2f8fd09487cccbd7097c6862
UE1_OP=11111111111111111111111111111111
```
**5) Start UERANSIM gNodeB.**
```
cd docker_open5gs
set -a
source .env
docker-compose -f nr-gnb.yaml up -d && docker attach nr_gnb
```
**6) Start UERANSIM UE.**
```
cd docker_open5gs
set -a
source .env
docker-compose -f nr-ue.yaml up -d && docker attach nr_ue
```
**7) Ping google.com**
In the 6) UE container, ping as follows.
```
ping google.com -I uesimtun0 -n
```

**Reply by herlesupreeth** [reply]
Thanks @s5uishida. @jni2000 i would try as @s5uishida mentioned if you dont have real gnb. 

But if you want to try out only 4G then I would suggest to use nsa-deploy.yaml file while deploying the core network in the above steps.
